subprojects/base-services/src/main/java/org/gradle/util/GFileUtils.java

```diff
@@ -316,4 +318,38 @@ public static void mkdirs(File dir) {
             throw new UncheckedIOException(String.format("Failed to create directory '%s'", dir));
         }
     }
+
+    /**
+     * Returns the path of target relative to base.
+     *
+     * @param target target file or directory
+     * @param base base directory
+     * @return the path of target relative to base.
+     */
+    public static String relativePathOf(File target, File base) {
```

Comment:
This could be replaced by [Path.relativize(other)](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#relativize(java.nio.file.Path)) if this code is only used with Java >= 7

Reply:
Thanks, I'll consider doing that in a subsequent PR. Also, this code came from our `FileResolver` implementation so it's been battle tested.